src/games/behavmixed.imp

```diff
@@ -36,149 +36,99 @@ template <class T>
+		prob = T(1) / T(num_actions);
       }
-      else if (GetSupport().Contains(node->GetInfoset()->GetAction(i)))
-	prob = (T) 1 / (T) GetSupport().NumActions(node->GetPlayer()->GetNumber(),
-						   node->GetInfoset()->GetNumber());
       else {
-	prob = (T) 0;
+		prob = T(0);
       }
     }
     else  {   // n.GetPlayer() == 0
-      prob = static_cast<T>(node->infoset->GetActionProb(i));
+      prob = T(node->infoset->GetActionProb(i));
     }
 
     GameTreeNodeRep *child = node->children[i];
-    m_bvals[child->number] = prob * m_bvals[node->number];
-    m_nvals[child->number] += m_bvals[child->number];
 
-    RealizationProbs(mp, pl, actions, child);
-  }
+    map_bvals[child] = prob * map_bvals[node];
+    map_nvals[child] += map_bvals[child];
 
+    RealizationProbs(mp, player, actions, child, map_nvals, map_bvals);
+  }
 }
 
 template <class T>
 MixedBehaviorProfile<T>::MixedBehaviorProfile(const MixedStrategyProfile<T> &p_profile)
   : m_probs(p_profile.GetGame()->NumActions()),
     m_support(p_profile.GetGame()),
-    m_gameversion(p_profile.GetGame()->GetVersion()),
-    m_cacheValid(false),
-    m_realizProbs(m_support.GetGame()->NumNodes()),
-    m_beliefs(m_support.GetGame()->NumNodes()),
-    m_nvals(m_support.GetGame()->NumNodes()),
-    m_bvals(m_support.GetGame()->NumNodes()),
-    m_nodeValues(m_support.GetGame()->NumNodes(),
-		 m_support.GetGame()->NumPlayers()),
-    m_infosetValues(m_support.GetGame()->NumInfosets()),
-    m_actionValues(m_support.GetGame()->NumActions()),
-    m_regret(m_support.GetGame()->NumActions())
-{
-  m_realizProbs = (T) 0.0;
-  m_beliefs = (T) 0.0;
-  m_nodeValues = (T) 0.0;
-  m_infosetValues = (T) 0.0;
-  m_actionValues = (T) 0.0;
-  m_regret = (T) 0.0;
-
-  ((Vector<T> &) *this).operator=((T)0);
+    m_gameversion(p_profile.GetGame()->GetVersion())
+{
+  static_cast<Vector<T> &>(m_probs) = T(0);
 
   GameTreeNodeRep *root = dynamic_cast<GameTreeNodeRep *>(m_support.GetGame()->GetRoot().operator->());
+
   const StrategySupportProfile &support = p_profile.GetSupport();
   GameRep *game = m_support.GetGame();
 
   for (auto player : game->GetPlayers()) {
-    m_nvals = (T) 0;
-    m_bvals = (T) 0;
-
+    std::map<GameNode,T> map_nvals, map_bvals;
     for (auto strategy : support.GetStrategies(player)) {
-      if (p_profile[strategy] > (T) 0) {
+      if (p_profile[strategy] > T(0)) {
         const Array<int> &actions = strategy->m_behav;
-        m_bvals[root->GetNumber()] = p_profile[strategy];
-        RealizationProbs(p_profile, player->GetNumber(), actions, root);
+        map_bvals[root] = p_profile[strategy];
+        RealizationProbs(p_profile, player, actions, root, map_nvals, map_bvals);
       }
     }
-
-    m_nvals[1] = (T) 1;   // set the root nval
-    BehaviorStrat(player->GetNumber(), root);
+    map_nvals[root] = T(1);   // set the root nval
+    auto root = m_support.GetGame()->GetRoot(); // BehaviorStrat takes GameNode not GameTreeNodeRep*
```

Comment:
The comment is not really necessary now - it only makes sense if you're looking at old versions of the code.